src/katgpucbf/utils.py

```diff
@@ -53,3 +58,48 @@ def parse_source(value: str) -> list[tuple[str, int]] | str:
         return [(ep.host, ep.port) for ep in endpoints]
     except ValueError:
         return value
+
+
+class DeviceStatus(Enum):
+    """Discrete `device-status` readings.
+
+    Doesn't follow the convention of :class:`aiokatcp.Sensor.Status` because
+    it's useful for `UNKNOWN` to have the highest (or worst) value.
+    """
+
+    OK = 1
+    DEGRADED = 2
+    FAIL = 3
+    UNKNOWN = 4
+
+
+class DeviceStatusSensor(aiokatcp.AggregateSensor):
+    """Summary sensor for quickly ascertaining device status.
+
+    This takes its value from the worst status of its target set of sensors, so
+    it's quick to identify if there's something wrong, or if everything is good.
+    """
+
+    def __init__(self, target: aiokatcp.SensorSet):
+        super().__init__(target=target, sensor_type=DeviceStatus, name="device-status")
+
+    def update_aggregate(
+        self,
+        updated_sensor: Optional[aiokatcp.Sensor[_T]],
+        reading: Optional[aiokatcp.Reading[_T]],
+        old_reading: Optional[aiokatcp.Reading[_T]],
+    ) -> aiokatcp.Reading[DeviceStatus]:  # noqa: D102
```

Comment:
```suggestion
    ) -> aiokatcp.Reading[DeviceStatus] | None:  # noqa: D102
```
For consistency with the base class.